patches/server/0739-Add-config-setting-for-logging-player-ip-addresses.patch

```diff
@@ -0,0 +1,93 @@
++    }
++
+     public static int itemValidationDisplayNameLength = 8192;
+     public static int itemValidationLocNameLength = 8192;
+     public static int itemValidationLoreLineLength = 8192;
+diff --git a/src/main/java/net/minecraft/network/protocol/PacketUtils.java b/src/main/java/net/minecraft/network/protocol/PacketUtils.java
+index bcf53ec07b8eeec7a88fb67e6fb908362e6f51b0..335756e9b1f552ebe2873f49ee43cb4c91ea83db 100644
+--- a/src/main/java/net/minecraft/network/protocol/PacketUtils.java
++++ b/src/main/java/net/minecraft/network/protocol/PacketUtils.java
+@@ -38,10 +38,11 @@ public class PacketUtils {
+                     // Paper start
+                     catch (Exception e) {
+                         Connection networkmanager = listener.getConnection();
++                        String playerIP = com.destroystokyo.paper.PaperConfig.logPlayerIpAddresses ? String.valueOf(networkmanager.getRemoteAddress()) : ""; // Paper
+                         if (networkmanager.getPlayer() != null) {
+-                            LOGGER.error("Error whilst processing packet {} for {}[{}]", packet, networkmanager.getPlayer().getScoreboardName(), networkmanager.getRemoteAddress(), e);
++                            LOGGER.error("Error whilst processing packet {} for {}{}", packet, networkmanager.getPlayer().getScoreboardName(), playerIP.isEmpty() ? "" : "[" + playerIP + "]", e); // Paper
+                         } else {
+-                            LOGGER.error("Error whilst processing packet {} for connection from {}", packet, networkmanager.getRemoteAddress(), e);
++                            LOGGER.error("Error whilst processing packet {} for connection from {}", packet, playerIP, e); // Paper
+                         }
+                         TextComponent error = new TextComponent("Packet processing error");
+                         networkmanager.send(new ClientboundDisconnectPacket(error), (future) -> {
+diff --git a/src/main/java/net/minecraft/server/network/ServerConnectionListener.java b/src/main/java/net/minecraft/server/network/ServerConnectionListener.java
+index 961660f6f9e00b93252519e38b74c66c53388ed2..01da5acacb0bbe73578760c9bf8158b9d0f1c8da 100644
+--- a/src/main/java/net/minecraft/server/network/ServerConnectionListener.java
++++ b/src/main/java/net/minecraft/server/network/ServerConnectionListener.java
+@@ -199,7 +199,7 @@ public class ServerConnectionListener {
+                                 throw new ReportedException(CrashReport.forThrowable(exception, "Ticking memory connection"));
+                             }
+ 
+-                            ServerConnectionListener.LOGGER.warn("Failed to handle packet for {}", networkmanager.getRemoteAddress(), exception);
++                            ServerConnectionListener.LOGGER.warn("Failed to handle packet for {}", com.destroystokyo.paper.PaperConfig.logPlayerIpAddresses ? networkmanager.getRemoteAddress() : "", exception);
```

Comment:
I think instead of an empty string, you should put like `<ip address withheld>`. Having people see `Failed to handle packet for ` in their logs might be a little confusing.